flutter/shell/platform/tizen/external_texture.h

```diff
@@ -13,36 +13,51 @@
 
 namespace flutter {
 
+static std::atomic<int64_t> next_texture_id = {1};
+
+class ExternalTexture {
+ public:
+  ExternalTexture() : texture_id_(next_texture_id++) {}
+
+  virtual ~ExternalTexture() = default;
+
+  // Returns the unique id for the ExternalTextureGL instance.
+  int64_t TextureId() { return texture_id_; }
+
+ protected:
+  const int64_t texture_id_ = 0;
+};
+
 enum class ExternalTextureExtensionType { kNone, kNativeSurface, kDmaBuffer };
 
 struct ExternalTextureGLState {
   uint32_t gl_texture;
   ExternalTextureExtensionType gl_extension;
 };
 
-static std::atomic<int64_t> next_texture_id = {1};
-
-class ExternalTexture {
+class ExternalGLTexture : public ExternalTexture {
  public:
-  ExternalTexture(ExternalTextureExtensionType gl_extension =
-                      ExternalTextureExtensionType::kNone)
-      : state_(std::make_unique<ExternalTextureGLState>()),
-        texture_id_(next_texture_id++) {
+  ExternalGLTexture(ExternalTextureExtensionType gl_extension =
+                        ExternalTextureExtensionType::kNone)
+      : ExternalTexture(), state_(std::make_unique<ExternalTextureGLState>()) {
     state_->gl_extension = gl_extension;
   }
 
-  virtual ~ExternalTexture() = default;
-
-  // Returns the unique id for the ExternalTextureGL instance.
-  int64_t TextureId() { return texture_id_; }
-
-  virtual bool PopulateTexture(size_t width,
-                               size_t height,
-                               FlutterOpenGLTexture* opengl_texture) = 0;
+  virtual bool PopulateOpenGLTexture(size_t width,
```

Comment:
I just wonder..
ExternalVulkanTexture -> PopulateVulkanTexture
ExternalGLTexture -> PopulateOpenGLTexture
Why not PopulateGLTexture ?

Reply:
Please check `FlutterTizenTextureRegistrar::PopulateOpenGLTexture()` too.